docs/interface.md

```diff
@@ -48,24 +56,81 @@ When the status is not `error`, either set the value to `null` or omit the key e
 
 #### Statuses
 
+> key: `status`
+
 The following per-test statuses are valid:
 - `pass`: The test passed
 - `fail`: The test failed
 - `error`: The test errored
 
 #### Messages
 
-The per-test `message` key can be used to display human-readable error messages. Presume that whatever is written here will be displayed to the student. If there is no error message, either set the value to `null` or omit the key entirely.
+> key: `message`
+
+The per-test `message` key can be used to display human-readable error messages. Presume that whatever is written here will be displayed to the student. If there is no error message, either set the value to `null` or omit the key entirely. It is also permissible to output test suite output here.
 
 #### Output
 
+> key: `output`
+
 The per-test `output` key should be used to store and output anything that a user deliberately outputs for a test.
 
 - It should be attached to all test results that produce user output.
 - Only content outputted by a user manually should show - not automatic output by the test-runner.
 - You may either capture content that is output through normal means (e.g. `puts` in Ruby, `print` in Python or `Debug.WriteLine` in C#), or you may provide a method that the user may use (e.g. the Ruby Test Runner provides a user with a globally available `debug` method that they can use, which has the same characteristics as the standard `puts` method).
 - The output **must** be limited to 500 chars. Either truncating with a message of "Output was truncated. Please limit to 500 chars" or returning an error in this situation are acceptible.
 
+#### Command
+
+> key: `cmd`
+
+This is the command that is being tested, apart from any test-suite-specific function and/or special syntax. The purpose of this is to clearly state what code is evaluated and subsequently failing the test.
+
+#### Expected
+
+> key: `expected`
+
+This is the value that is expected from the command (see `"cmd"` key) being tested, apart from any test-specific functions and/or special syntax. The purpose of this is to clearly state what value is expected when the command is evaluated.
+
+## Why have separate command and expected keys
+
+The purpose of Exercism is to teach programming language fluency. Testing is a specific concept all to itself in each language: how it is performed, what strategies to use, best-practices, etc. Requiring students to learn the test framework to understand very early exercises is counter-intuitive to the approach of language _fluency_.
+
+All this to say, the separate keys exist to facilitate a very plain, human approach to the test suite for each language. It facilitates the website UI/UX, which may or may not show the test suite to the student.
+
+### UI/UX concerns
+
+#### On test failure
+
+When a student's solution fails a test, it should display something like:
+
+```text
+We ran:
+  <cmd>
+
+It resulted in:
+  <message>
+```
```

Comment:
I would have expected the "expected" value to be shown here. Or am I mistaken @iHiD?